main.go

```diff
@@ -74,13 +48,32 @@ func (hc *HealthCheckWebService) apiGetIsReplica(w http.ResponseWriter, r *http.
 		panic(err)
 	}
 
+	// if byte lag exceeds max_allowable_byte_lag then return 500
+	if hc.maxAllowableByteLagExceeded(r, nodeInfo) {
```

Comment:
Should we move this after the `!IsReplica` check first to ensure it's not a primary?

Reply:
yes!